pypeit/core/skysub.py

```diff
@@ -742,6 +744,10 @@ def local_skysub_extract(sciimg, sciivar, tilts, waveimg, global_sky, thismask,
             img_minsky = sciimg - skyimage
             for ii in range(objwork):
                 iobj = group[ii]
+                # boxcar radius
```

Comment:
I'm not convinced that we need a different box_rad for the maskdef sources. It may be that we need to think through some subtle issues for faint sources such that this box_rad parameters should be an array input rather than a single one-size fits all scalar. However, let's not key up on the maskdef_extract. This is a core method, which should just receive general instructions about boxcar radii, not specific instructions that depend on some upstream functions/tags being set in sobjs. 

So let's discuss whether a flux dependent boxcar radius is necessary in this routine (I'm not convinced that it is yet), but if so let's just make the box_rad an array/or scalar input and parse the maskdef stuff uprstream from this routine (i.e. in reduce for example, or just make this an  attribute of sobjs, i.e. it takes a default box_rad but then modifies it if maskdef is set).

For what it is worth X says he may have suggested this modification to box_rad but cannot remember why.